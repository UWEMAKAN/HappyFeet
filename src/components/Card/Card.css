.Card {
  height: 40rem;
  width: 100%;
  align-self: center;
  justify-self: center;
  padding: 0 2rem;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
}

.Title {
  font-weight: 300;
  color: #000000;
  align-self: flex-start;
  margin: 1rem 0;
}
