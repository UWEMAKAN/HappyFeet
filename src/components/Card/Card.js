import React from 'react';
import propTypes from 'prop-types';
import Button from '../Common/Button/Button';
import Image from '../Common/Image/Image';
import Rating from './Rating/Rating';
import './Card.css';
import Pricing from './Pricing/Pricing';

const Card = ({
  title, image, price, discount, rating, clickHandler
}) => (
  <div className="Card">
    <Image title={title} image={image} />
    <Button text="Buy" styles={{ marginRight: '1rem' }} clickHandler={clickHandler} />
    <h3 className="Title">{title}</h3>
    <Rating rating={rating} />
    <Pricing price={price} discount={discount} />
  </div>
);

Card.propTypes = {
  title: propTypes.string.isRequired,
  image: propTypes.string.isRequired,
  price: propTypes.number.isRequired,
  discount: propTypes.number.isRequired,
  rating: propTypes.number.isRequired,
  clickHandler: propTypes.func.isRequired
};

export default Card;
