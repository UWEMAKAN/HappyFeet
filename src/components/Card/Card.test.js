import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

const renderCard = (args) => {
  const defaultProps = {
    title: '',
    image: '',
    price: 0,
    discount: 0,
    rating: 0,
    clickHandler: jest.fn()
  };
  const props = { ...defaultProps, ...args };
  // eslint-disable-next-line react/jsx-props-no-spreading
  return shallow(<Card {...props} />);
};

describe('Card Component', () => {
  const title = 'Black Leather Shoes';
  const image = 'http://localhost:3001/assets/images/image.jpg';
  const price = 4000;
  const discount = 30;
  const rating = 4.6;
  const clickHandler = jest.fn();

  test('should render Card', () => {
    const wrapper = renderCard({
      title, image, price, discount, rating, clickHandler
    });
    expect.assertions(2);
    expect(wrapper.getElements()[0].props.children.length).toBe(5);
    expect(wrapper).toMatchSnapshot();
  });
});
