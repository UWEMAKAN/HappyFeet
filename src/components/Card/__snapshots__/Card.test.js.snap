// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should render Card 1`] = `ShallowWrapper {}`;
